File: back-end/src/main/java/pl/straburzynski/campfiresongs/song/exception/JsonParseException.java
package pl.straburzynski.campfiresongs.song.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;
import pl.straburzynski.campfiresongs.exception.CustomException;
import pl.straburzynski.campfiresongs.exception.ErrorResponse;

@ResponseStatus(HttpStatus.BAD_REQUEST)
public class JsonParseException extends CustomException {

    public JsonParseException(String methodName) {
        super(ErrorResponse.builder()
                .message("JSON parse exception on method: " + methodName)
                .build());
    }
}

File: back-end-kotlin/src/main/kotlin/pl/straburzynski/campfiresongs/song/exception/JsonParseException.kt
package pl.straburzynski.campfiresongs.song.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus
import pl.straburzynski.campfiresongs.exception.CustomException
import pl.straburzynski.campfiresongs.exception.ErrorResponse

@ResponseStatus(HttpStatus.BAD_REQUEST)
class JsonParseException(methodName: String) : CustomException(
    ErrorResponse(
        emptyMap(),
        "exception.json_parse_exception",
        "JSON parse exception on method: $methodName"
    )
)